pkg/kapis/devops/v1alpha4/register.go

```diff
@@ -0,0 +1,45 @@
+package v1alpha4
+
+import (
+	"github.com/emicklei/go-restful"
+	"kubesphere.io/devops/pkg/api"
+	"kubesphere.io/devops/pkg/api/devops/v1alpha4"
+	"kubesphere.io/devops/pkg/apiserver/runtime"
+	"net/http"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// Option holds some useful tools when creating web service.
+type Option struct {
+	Container *restful.Container
+	Client    client.Client
+}
+
+// AddToContainer registers web services with some options.
+func AddToContainer(o Option) {
+	// create API handler
+	h := newAPIHandler(apiHandlerOption{
+		client: o.Client,
+	})
+	// create web service without group
+	ws := runtime.NewWebServiceWithoutGroup(v1alpha4.GroupVersion)
+	addToContainer(o, ws, h)
+	// create web service with group
+	ws = runtime.NewWebService(v1alpha4.GroupVersion)
+	addToContainer(o, ws, h)
+}
+
+func addToContainer(o Option, ws *restful.WebService, handler *apiHandler) {
+	ws.Route(ws.GET("/namespaces/{namespace}/pipelines/{pipeline}/pipelineruns").
+		To(handler.listPipelineRuns()).
+		Doc("Get all runs of the specified pipeline").
+		Param(ws.PathParameter("namespace", "Namespace of the pipeline")).
+		Param(ws.PathParameter("pipeline", "Name of the pipeline")).
+		Param(ws.QueryParameter("branch", "The name of SCM reference")).
+		Param(ws.QueryParameter("backward", "Backward compatibility. Default value: true").
```

Comment:
Please explain the compatibility. It's hard to understand if users try to read the API document which was generated by this.